python/pyspark/accumulators.py

```diff
@@ -109,10 +109,14 @@
 
 def _deserialize_accumulator(aid, zero_value, accum_param):
     from pyspark.accumulators import _accumulatorRegistry
-    accum = Accumulator(aid, zero_value, accum_param)
-    accum._deserialized = True
-    _accumulatorRegistry[aid] = accum
-    return accum
+    # If this certain accumulator was deserialized, don't overwrite it.
+    if aid in _accumulatorRegistry:
```

Comment:
Ah, so the problem is this accumulator is de/serialized multiple times and `_deserialize_accumulator` modifies the global status multiple times. I see. LGTM.

Reply:
Yes.